src/myr/Myr.js

```diff
@@ -1047,6 +1053,164 @@ class Myr {
         return outerElId;
     }
 
+    setSeed = (seed = 0) => {
+        if (seed === 0){
+            let newSeed = new Date().getTime();
+            while(this.rand.seed === newSeed)
+            {
+                newSeed = new Date().getTime();
+            }
+            this.rand.seed = newSeed;
+        }
+        else{
+            this.rand.seed = seed;
+        }
+        this.rand.randCounter = 0;
+    }
+
+    getSeed = () => {
+        return this.rand.seed;
+    }
+
+    getSeedCounter = () => {
+        return this.rand.randCounter;
+    }
+
+    setSeedCounter = (counter = 0) => {
+        this.rand.randCounter = counter;
+    }
+
+    decrementRandCounter = () => {
+        this.rand.randCounter -= 1;
+    }
+
+    calculateRandom = (seedNum, nextRand) => {
+        let small = seedNum*0.000000000000000001;
+        let counterPow = this.rand.randCounter * this.rand.randCounter;
+        return (seedNum*Math.PI*counterPow*small*1.543637*9.479137*0.5*nextRand);
+    }
+
+    randomInt = (min = -40, max = 40) => {
+        if(max < min)
+        {
+            let temp = max;
+            min = max;
+            max = temp;
+        }
+        if(max === min)
+        {
+            return max;
+        }
+
+        let range = max - min;
+        let randNum, seedNum, nextRand;
+
+        if(this.rand.seed === 0)
```

Comment:
Is it ever possible to set the seed to 0? If not, then you don't need this check.

Reply:
The seed is first initialized to 0 in the constructor, which is also why several functions check if its 0 instead of uninitialized.

Reply:
This goes against JS best practices. It shouldn't be initialized to 0 if 0 isn't a valid value. It should be left as undefined and then you can check if it's undefined or not.